docs/FLUTTER_STYLE.md

```diff
@@ -0,0 +1,86 @@
+# Code Style
+
+All rules fall through to the [Effective Dart](https://dart.dev/guides/language/effective-dart)
+guidelines.
+
+## 1. All comments, identifiers, issues, etc. should be written in English
+
+## 2. Limit code to contain 80 characters per line
+
+Exceptions:
+
+- documentation code
+- hard-coded URLs (e.g., in comments)
+- TODO comments should stay on a single line, so that tools can read & report
+  the full message
+
+## 3. Single- vs. Multi-line function calls
+
+Put calls in a single line if they fit:
+
+```dart
+return Container(color: Colors.pink, child: myChild);
+```
+
+Wrap if there are more than two named parameters, they get longer:
+
+```dart
+return Container(
+  color: context.theme.primaryColor,
+  padding: EdgeInsets.symmetric(horizontal: 16),
+  child: myChild,
+);
+```
+
+## 4. Always prefer relative to absolute imports
+
+E.g. `import '../common/module.dart'` instead of `import '/common/module.dart'`
+
+## 5. Use private functions/classes if they are only used locally
+
+In Flutter, you make symbols private by prefixing them with an underscore (e.g. `_DatePickerState`).
+
+## 6. Group concepts within a file by their functionality
+
+## 7. Order these groups of concepts within a file by their relevance
+
+E.g. Put helper methods beneath the methods they help
+
+Why? The name of helper methods should convey their function so you don't need
+to view their contents to get an initial understanding of the class.
+
+## 8. Mark constructors as `const` where possible
+
+## 9. Prefer using named arguments for functions with multiple arguments
+
+## 10. Anonymous functions
```

Comment:
What about lambdas that span over more than one line? Consider the following where I have a long function name / something that gets wrapped to the next line by the formatter

```dart
int foo(Bar b) =>
  reallyLongFunctionNameThatGetsWrappedToNextLine(b).toInt()
```
I think it would make sense to touch on this in the section seeing as this has come up once or twice already in code reviews. IIFC you suggested that if it gets wrapped to a new line to add curly braces, resulting in 

```dart
int foo(Bar b) => {
  reallyLongFunctionNameThatGetsWrappedToNextLine(b).toInt()
}

// if it fits on one line without getting auto-wrapped, omit braces
String baz(Bar b) => shortFunction(b)
```

I would also agree with this.